src/main/java/bio/terra/service/dataset/flight/upgrade/predictableFileIds/ConvertToPredictableFileIdsUpdateMetadataStep.java

```diff
@@ -0,0 +1,26 @@
+package bio.terra.service.dataset.flight.upgrade.predictableFileIds;
+
+import bio.terra.service.dataset.DatasetService;
+import bio.terra.service.job.DefaultUndoStep;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import java.util.UUID;
+
+public class ConvertToPredictableFileIdsUpdateMetadataStep extends DefaultUndoStep {
```

Comment:
Aha, it actually doesn't matter since it runs in a transaction.  Though I'm realizing I didn't set the rollback option to true in the flight.  Will do that

Reply:
Er rather, it doesn't roll back so it just doesn't show up.  Either way, we should be ok

Reply:
Ohhhh!  Okay, sounds good!

When I see a step that doesn't clean up after itself because the clean up is handled in some way elsewhere in the flight, I find it helpful to have a comment to that effect. If you wanted to unit test this, I'd check the flight construction to make sure that this step is scheduled after a transaction opened step.  But I won't hold up my approval, these are thoughts more applicable to enduring code.